### PR TITLE
Issue 81: touch up XML some

### LINKED ISSUE #73: As a n00b paginator, there might be an off-by-1 error in the `limit` parameter
## 🐛 Describe the bug

The API at https://pds-gamma.jpl.nasa.gov/api/ has pagination features that include a start index and a limit described as "maximum number of matching results returned, for pagination". This is also reflected in the Swagger definition and the [draft specification document](https://docs.google.com/document/d/16d0MVh48bFLvWsa5-B_Hy-cby1rGWdnNojWOJpUcOvA/edit#).

For the `/collections/{lidvid}/products` endpoint, this is the case. But for the `/bundles/{lidvid}/collections` it returns 1 more than the limit.

## 📜 To Reproduce

Run `curl --silent --header 'Accept: application/json' 'https://pds-gamma.jpl.nasa.gov/api/collections/urn%3Anasa%3Apds%3Ainsight_documents%3Adocument_hp3rad%3A%3A8.0/products?start=0&limit=0&fields=ops%3AData_File_Info.ops%3Amd5_checksum&only-summary=false' | json_pp` and note that the limit is 0 and 0 results are returned in `data` ✅. `limit=1` gives 1, `limit=2` gives 2.

But compare that with `curl --silent --header 'Accept: application/json' 'https://pds-gamma.jpl.nasa.gov/api/bundles/urn%3Anasa%3Apds%3Ainsight_documents%3A%3A2.0/collections?start=0&limit=0&fields=ops%3AData_File_Info.ops%3Amd5_checksum&only-summary=false' | json_pp` and note that the limit is 0 but 1 result is the `data` is returned ❌. `limit=1` gives 2, `limit=2` gives 3, etc. `limit=-1` gives 0! 

## 🕵️ Expected behavior

Consistency in `limit` between endpoints.

## 📚 Version of Software Used

`[ base url: /api/ , api version: 0.0.1 ]`

## 🖥 System Info
- OS: independent
- Browser: independent
- Version: 0.0.1

---

** :unicorn: Applicable requirements**

### LINKED ISSUE #73: blob/json_blob should not be included in default response
## 🐛 Describe the bug

I've seen the blob in default responses from the API

## 📜 To Reproduce
Test with this request for example:

curl --location --request GET 'http://localhost:8080/bundles' \
--header 'Accept: application/kvp+json'

## 🕵️ Expected behavior

The blob should not be available in default responses so to improve the performance and not to confuse the users.

## 📚 Version of Software Used

Latest on main

### LINKED ISSUE #81: [pds-api] B12.0 Improve API query handling

### LINKED ISSUE #81: Service using JSON blob in pds4+xml response when it should use ops blob
## 🐛 Describe the bug

For `curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0' --header 'Accept: application/pds4+xml'`

The `<pds4>` should be XML but JSON is being decoded:

```
<pds4>{"Product_Document":{"Identification_Area":{"product_class":"Product_Document","Modification_History":{"Modification_Detail":{"modification_date":"2016-03-16","description":"Initial PDS4 version.","version_id":1}},"information_model_version":"1.14.0.0","logical_identifier":"urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst","version_id":1,"Citation_Information":{"publication_year":2016,"description":"Description of the MESSENGER Neutron Spectrometer (NS)."},"title":"MESSENGER Neutron Spectrometer (NS) Description"},"xmlns":"http://pds.nasa.gov/pds4/pds/v1","xsi:schemaLocation":"http://pds.nasa.gov/pds4/pds/v1 http://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1E00.xsd","xmlns:xsi":"http://www.w3.org/2001/XMLSchema-instance","Context_Area":{"Investigation_Area":{"Internal_Reference":{"lid_reference":"urn:nasa:pds:context:investigation:mission.messenger","reference_type":"document_to_investigation"},"name":"MESSENGER","type":"Mission"},"Observing_System":{"name":"MESSENGER","Observing_System_Component":[{"Internal_Reference":{"lid_reference":"urn:nasa:pds:context:instrument_host:spacecraft.mess","reference_type":"is_instrument_host"},"name":"MESSENGER","type":"Host"},{"Internal_Reference":{"lid_reference":"urn:nasa:pds:context:instrument:ns.mess","reference_type":"is_instrument"},"name":"NS","type":"Instrument"}]},"Target_Identification":{"Internal_Reference":{"lid_reference":"urn:nasa:pds:context:target:planet.mercury","reference_type":"document_to_target"},"name":"Mercury","type":"Planet"}},"Document":{"publication_date":"2016-03-16","description":"Description of the MESSENGER Neutron Spectrometer (NS).","Document_Edition":{"files":1,"language":"English","Document_File":{"file_name":"ns_inst.pdf","document_standard_id":"PDF/A"},"edition_name":"MESSENGER NS Instrument Description"}}}}</pds4>
```

## 📜 To Reproduce
Using curl above the `<pds4>` element decodes to JSON.

## 🕵️ Expected behavior
Using curl from above the `<pds4>` element decodes to XML.

## 🖥 System Info

---

### LINKED ISSUE #155: `products/{identifier}` missing properties object in application/json response
## 🐛 Describe the bug

missing properties object in application/json response (maybe others?)

## 📜 To Reproduce
When querying `products/` endpoint, I get:
```json
...
  "data": [
    {
      "id": "urn:nasa:pds:orex.ovirs::10.0",
      "type": "Product_Bundle",
      "title": "Origins, Spectral Interpretation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx): Visible and InfraRed Spectrometer (OVIRS) Bundle",
      "start_date_time": "2016-09-08T23:05:00Z",
      "stop_date_time": "3000-01-01T00:00:00.000Z",
      "investigations": [
        {
          "id": "urn:nasa:pds:context:investigation:mission.orex",
          "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:investigation:mission.orex"
        }
      ],
      "observing_system_components": [
        {
          "id": "urn:nasa:pds:context:instrument_host:spacecraft.orex",
          "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:instrument_host:spacecraft.orex"
        },
        {
          "id": "urn:nasa:pds:context:instrument:ovirs.orex",
          "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:instrument:ovirs.orex"
        }
      ],
      "targets": [
        {
          "id": "urn:nasa:pds:context:target:asteroid.101955_bennu",
          "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:target:asteroid.101955_bennu"
        }
      ],
      "metadata": {
        "update_date_time": "2018-05-31T00:00:00Z",
        "version": "10.0",
        "label_url": "/data/pds4/test-data/registry/orex.ovirs/bundle_ovirs.xml"
      },
      "properties": {
        "pds:Modification_Detail.pds:description": [
          "Creation. This Product_Bundle label was written by OREX personnel, for the initial draft ovirs bundle.",
        ...
        ],
        "pds:Target_Identification.pds:alternate_designation": [
          "1999 RQ36"
        ],
...
```

But with `products/urn:nasa:pds:orex.ovirs::10.0` (missing properties):
```json
{
  "id": "urn:nasa:pds:orex.ovirs::10.0",
  "type": "Product_Bundle",
  "title": "Origins, Spectral Interpretation, Resource Identification, Security, Regolith Explorer (OSIRIS-REx): Visible and InfraRed Spectrometer (OVIRS) Bundle",
  "start_date_time": "2016-09-08T23:05:00Z",
  "stop_date_time": "3000-01-01T00:00:00.000Z",
  "investigations": [
    {
      "id": "urn:nasa:pds:context:investigation:mission.orex",
      "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:investigation:mission.orex"
    }
  ],
  "observing_system_components": [
    {
      "id": "urn:nasa:pds:context:instrument_host:spacecraft.orex",
      "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:instrument_host:spacecraft.orex"
    },
    {
      "id": "urn:nasa:pds:context:instrument:ovirs.orex",
      "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:instrument:ovirs.orex"
    }
  ],
  "targets": [
    {
      "id": "urn:nasa:pds:context:target:asteroid.101955_bennu",
      "href": "http://pds-gamma.jpl.nasa.gov/api/products/urn:nasa:pds:context:target:asteroid.101955_bennu"
    }
  ],
  "metadata": {
    "update_date_time": "2018-05-31T00:00:00Z",
    "version": "10.0",
    "label_url": "/data/pds4/test-data/registry/orex.ovirs/bundle_ovirs.xml"
  }
}
```

## 🕵️ Expected behavior

`properties` block should be included in `products/{identifier}` response as well.

---

## ⚙️ Engineering Details
We should definitely test the other endpoints to make sure this is working correctly as well.

### LINKED ISSUE #456: As an API user, I want to get an XML response
## Motivation

...so that the API can support numerous format responses to support the community.

## Details

**Note: The content should match the application/json structure**

That can be done easily by applying the default springboot content negotiation instead of the custom made content negotiation which returns the original pds4 label.

The pds4 label response could be kept for content-type=application/pds4+xml

## Acceptance criteria

**Given** any API request
**When I perform** with content-type header 'application/xml'
**I expect to** get the same attributes and same structure as the same request run with content-type header set to `application/json`

### PR DESCRIPTION
**Pull request by al-niessner**
## 🗒️ Summary
Cleaned up more code. Handle the code more or less uniformly from endpoint to return result.

Update also appears to fix a bug identified here: https://github.com/NASA-PDS/pds-api/issues/155

## ⚙️ Test Data and/or Report

```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0' --header 'Accept: application/pds4+xml'
```

## ♻️ Related Issues

Resolves NASA-PDS/pds-api#81 
Resolves NASA-PDS/pds-api#73 (update to newest dataset exposed this and then fixing pds4 blob fixed this as well)
Resolves https://github.com/NASA-PDS/pds-api/issues/155
Resolves https://github.com/NASA-PDS/registry-api/issues/456

### CODE REVIEW

**Comment by al-niessner** [comment]
Does the decode/inflate of the blob to string of XML but now the serializer makes a mess of the XML string doing substitution for all of the XML special characters like '<'.

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/Pds4XmlProductSerializer.java [hunk 1]
```diff
@@ -18,42 +19,56 @@
...
+					Pds4XmlProductSerializer.NAMESPACE_URL);
+			writer.writeStartElement(Pds4XmlProductSerializer.NAMESPACE_URL, "products");
+			writer.writeDefaultNamespace("http://pds.nasa.gov/pds4/pds/v1");
+			writer.writeNamespace(Pds4XmlProductSerializer.NAMESPACE_PREFIX, 
+	        		  Pds4XmlProductSerializer.NAMESPACE_URL);
+			Pds4XmlProductSerializer.serialize(writer, new XmlMapper(), product);
+		}
+		catch (ClassCastException e)
+		{ this.logger.error("For XML serialization, the Product object must be extended as ProductWithXmlLabel: " + e.getMessage()); }
+		catch (Exception e) {}
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/Pds4XmlProductSerializer.java]: fixed

File: service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java [hunk 2]
```diff
@@ -280,4 +280,18 @@ public void setResponse(RestHighLevelClient client, SearchRequest request) throw
...
         this.setResponse(client.search(request, RequestOptions.DEFAULT).getHits());
 	}
+
+	public void setSingularResponse(RestHighLevelClient client, SearchRequest request) throws IOException
+	{
+        request.source().size(this.getLimit());
+        request.source().from(this.getStart());
+        SearchHits hits = client.search(request, RequestOptions.DEFAULT).getHits();
+        
+        if (hits.getTotalHits().value == 1L) this.formatters.get(this.format).setResponse(hits.getAt(0), this.fields);
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/elasticsearch/business/RequestAndResponseContext.java]: fixed

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Can you look at the output for the XML types please? I am sure it is not correct but I need to know what tags need fixed and how. The biggest problem is blob needs to be inserted as XML and that is just causing jackson converter to go nuts. I have something working now, but it is less than perfect. I am not trying to reach perfect nor walk the perfect path but just to deliver XML that is good enough.

File: service/src/main/java/gov/nasa/pds/api/engineering/serializer/Pds4XmlProductSerializer.java [hunk 3]
```diff
@@ -18,42 +19,62 @@
...
+	static public void serialize (OutputStream stream, XMLStreamWriter writer, XmlMapper mapper, Pds4Product product) throws IOException, XMLStreamException
+	{
+		writer.writeStartElement("id");
+		writer.writeCharacters(product.getId());
+		writer.writeEndElement();
+		mapper.writeValue(writer, product.getMetadata());
+		writer.writeStartElement("pds4");
+		writer.writeCharacters(" ");
+		writer.flush();
+		stream.write(String.valueOf (product.getPds4()).getBytes());
```
[the remark this answers is not in the document]
Comment (al-niessner) [reply service/src/main/java/gov/nasa/pds/api/engineering/serializer/Pds4XmlProductSerializer.java]: fixed

**Comment by tloubrieu-jpl** [comment]
Thanks @al-niessner 

I looked at the new output and that looks good overall.

For the details I tried to see how the xmlns prefix should be used and provided some notes on that. I also don't understand why Pds4Metadata tag became the new name of meta.

Here are the details of my notes:
Using request:
```
curl --location --request GET 'http://localhost:8080/products' \
--header 'Accept: application/pds4+xml'
```

1. I don't think we need the first `
```
xmlns="http://pds.nasa.gov/pds4/pds/v1"` 
```
since it relates to the pds4 schema which is already in the pds4 tag below.

2. pds4metadata
'
```
  <id>urn:nasa:pds:insight_rad:data_raw:hp3_rad_raw_00390_20200101_120222::1.0</id>
            <Pds4Metadata xmlns="">
```

The id should have a pds_api tag prefix since it relates to the pds_api

I would keep the pds_api:meta tag as it is for pds4+json

We should not have an ampty xmlns attribute.


3. Could we have the embedded data_files tag singular (without 's') ? I know it is not always easy with the existing marshallers or unmarshallers.
```
<data_files>
    <data_files>
```

4. the pds4 decoded label looks good. pds4 tag should prefixed with pds_api.


Using request:
```
curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:insight_rad:data_raw:hp3_rad_raw_00390_20200101_120222::1.0' \
--header 'Accept: application/pds4+xml'
```

1. I believe all this section
```
  <id>urn:nasa:pds:insight_rad:data_raw:hp3_rad_raw_00390_20200101_120222::1.0</id>
    <Pds4Metadata xmlns="">
        <node_name>PDS_ENG</node_name>
        <label_file>
            <file_name>hp3_rad_raw_00390_20200101_120222.xml</file_name>
            <file_ref>/data/data_raw/hp3_rad_raw_00390_20200101_120222.xml</file_ref>
            <creation_date>2020-06-02T17:58:34Z</creation_date>
            <file_size>23809</file_size>
            <md5_checksum>5317a6bbadca248da550780f37d531df</md5_checksum>
        </label_file>
        <data_files>
            <data_files>
                <file_name>hp3_rad_raw_00390_20200101_120222.tab</file_name>
                <file_ref>/data/data_raw/hp3_rad_raw_00390_20200101_120222.tab</file_ref>
                <creation_date>2020-06-01T11:36:37Z</creation_date>
                <file_size>21216107</file_size>
                <md5_checksum>c1f92f8eb97f437599545263ecf113dd</md5_checksum>
                <mime_type>text/plain</mime_type>
            </data_files>
        </data_files>
    </Pds4Metadata>
```
Should have a `pds_api:` prefix

`Pds4Metadata` tag name should be meta

2. pds4 tag name should also have the pds_api: prefix

**Comment by jordanpadams** [comment]
@tloubrieu-jpl @al-niessner for this one, let's table it:

> Could we have the embedded data_files tag singular (without 's') ? I know it is not always easy with the existing marshallers or unmarshallers.

I would like to create a new ticket to rename this section altogether to use the `ops:` namespace

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

I left the xmlns in the `<product>` because I do not understand the reason to remove them. If the pds_api namespace is defined by `xmlns:pds_api="http://pds.nasa.gov/api" xmlns="http://pds.nasa.gov/pds4/pds/v1"` then it should remain with `<product>` and the ones in `<Product_Document>` are redundant but harmless.

The double usage is probably because the namespace is required when the blob is taken standalone. Therefore, if you want aesthetic results, then I would remove the ones from the decoded blob since they are technically the redundant definitions. Note: this will be a huge pain to actually do unless you can guarantee that they are all identical in all blobs.

**Comment by al-niessner** [comment]
@tloubrieu-jpl 

I added and modified tags are requested. Here is the result but also pushed so feel free to run your own tests:

```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0' --header 'Accept: application/pds4+xml' | xmllint --format -
<?xml version="1.0"?>
<pds_api:product xmlns:pds_api="http://pds.nasa.gov/api" xmlns="http://pds.nasa.gov/pds4/pds/v1">
  <pds_api:id>urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0</pds_api:id>
  <pds_api:meta>
    <node_name>PSA</node_name>
    <label_file>
      <file_name>ns_inst.xml</file_name>
      <file_ref>/var/local/harvest/archive/document/ns_inst.xml</file_ref>
      <creation_date>2022-01-24T20:08:23Z</creation_date>
      <file_size>3589</file_size>
      <md5_checksum>a8d09cca0a01728db50c15052c2736cf</md5_checksum>
    </label_file>
    <data_files>
      <data_files>
        <file_name>ns_inst.pdf</file_name>
        <file_ref>/var/local/harvest/archive/document/ns_inst.pdf</file_ref>
        <creation_date>2022-01-24T20:08:23Z</creation_date>
        <file_size>138172</file_size>
        <md5_checksum>8103f20c13a3c321dac4a193aba19d16</md5_checksum>
        <mime_type>application/pdf</mime_type>
      </data_files>
    </data_files>
  </pds_api:meta>
  <pds_api:pds4>
    <Product_Document xmlns="http://pds.nasa.gov/pds4/pds/v1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pds.nasa.gov/pds4/pds/v1 http://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1E00.xsd">
      <Identification_Area>
        <logical_identifier>urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst</logical_identifier>
        <version_id>1.0</version_id>
        <title>MESSENGER Neutron Spectrometer (NS) Description</title>
        <information_model_version>1.14.0.0</information_model_version>
        <product_class>Product_Document</product_class>
        <Citation_Information>
          <publication_year>2016</publication_year>
          <description>Description of the MESSENGER Neutron Spectrometer (NS).</description>
        </Citation_Information>
        <Modification_History>
          <Modification_Detail>
            <modification_date>2016-03-16</modification_date>
            <version_id>1.0</version_id>
            <description>Initial PDS4 version.</description>
          </Modification_Detail>
        </Modification_History>
      </Identification_Area>
      <Context_Area>
        <Investigation_Area>
          <name>MESSENGER</name>
          <type>Mission</type>
          <Internal_Reference>
            <lid_reference>urn:nasa:pds:context:investigation:mission.messenger</lid_reference>
            <reference_type>document_to_investigation</reference_type>
          </Internal_Reference>
        </Investigation_Area>
        <Observing_System>
          <name>MESSENGER</name>
          <Observing_System_Component>
            <name>MESSENGER</name>
            <type>Host</type>
            <Internal_Reference>
              <lid_reference>urn:nasa:pds:context:instrument_host:spacecraft.mess</lid_reference>
              <reference_type>is_instrument_host</reference_type>
            </Internal_Reference>
          </Observing_System_Component>
          <Observing_System_Component>
            <name>NS</name>
            <type>Instrument</type>
            <Internal_Reference>
              <lid_reference>urn:nasa:pds:context:instrument:ns.mess</lid_reference>
              <reference_type>is_instrument</reference_type>
            </Internal_Reference>
          </Observing_System_Component>
        </Observing_System>
        <Target_Identification>
          <name>Mercury</name>
          <type>Planet</type>
          <Internal_Reference>
            <lid_reference>urn:nasa:pds:context:target:planet.mercury</lid_reference>
            <reference_type>document_to_target</reference_type>
          </Internal_Reference>
        </Target_Identification>
      </Context_Area>
      <Document>
        <publication_date>2016-03-16</publication_date>
        <description>Description of the MESSENGER Neutron Spectrometer (NS).</description>
        <Document_Edition>
          <edition_name>MESSENGER NS Instrument Description</edition_name>
          <language>English</language>
          <files>1</files>
          <Document_File>
            <file_name>ns_inst.pdf</file_name>
            <document_standard_id>PDF/A</document_standard_id>
          </Document_File>
        </Document_Edition>
      </Document>
    </Product_Document>
  </pds_api:pds4>
</pds_api:product>

```

**Comment by jordanpadams** [comment]
@al-niessner this should be revised:

```xml
<pds_api:product xmlns:pds_api="http://pds.nasa.gov/api" xmlns="http://pds.nasa.gov/pds4/pds/v1">
```
should be:

```xml
<pds_api:product xmlns:pds_api="http://pds.nasa.gov/api">
```

This namespace `http://pds.nasa.gov/pds4/pds/v1` should not apply at the top-level API response. As you noted, having this at each individual product as it comes back from the blob is fine.

**Comment by al-niessner** [comment]
@tloubrieu-jpl @jordanpadams 

Removed default namespace.

```
$ curl --location --request GET 'http://localhost:8080/products/urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0' --header 'Accept: application/pds4+xml' | xmllint --format -
<?xml version="1.0"?>
<pds_api:product xmlns:pds_api="http://pds.nasa.gov/api">
  <pds_api:id>urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst::1.0</pds_api:id>
  <pds_api:meta>
    <node_name>PSA</node_name>
    <label_file>
      <file_name>ns_inst.xml</file_name>
      <file_ref>/var/local/harvest/archive/document/ns_inst.xml</file_ref>
      <creation_date>2022-01-24T20:08:23Z</creation_date>
      <file_size>3589</file_size>
      <md5_checksum>a8d09cca0a01728db50c15052c2736cf</md5_checksum>
    </label_file>
    <data_files>
      <data_files>
        <file_name>ns_inst.pdf</file_name>
        <file_ref>/var/local/harvest/archive/document/ns_inst.pdf</file_ref>
        <creation_date>2022-01-24T20:08:23Z</creation_date>
        <file_size>138172</file_size>
        <md5_checksum>8103f20c13a3c321dac4a193aba19d16</md5_checksum>
        <mime_type>application/pdf</mime_type>
      </data_files>
    </data_files>
  </pds_api:meta>
  <pds_api:pds4>
    <Product_Document xmlns="http://pds.nasa.gov/pds4/pds/v1" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://pds.nasa.gov/pds4/pds/v1 http://pds.nasa.gov/pds4/pds/v1/PDS4_PDS_1E00.xsd">
      <Identification_Area>
        <logical_identifier>urn:nasa:pds:izenberg_pdart14_meap:document:ns_inst</logical_identifier>
        <version_id>1.0</version_id>
        <title>MESSENGER Neutron Spectrometer (NS) Description</title>
        <information_model_version>1.14.0.0</information_model_version>
        <product_class>Product_Document</product_class>
        <Citation_Information>
          <publication_year>2016</publication_year>
          <description>Description of the MESSENGER Neutron Spectrometer (NS).</description>
        </Citation_Information>
        <Modification_History>
          <Modification_Detail>
            <modification_date>2016-03-16</modification_date>
            <version_id>1.0</version_id>
            <description>Initial PDS4 version.</description>
          </Modification_Detail>
        </Modification_History>
      </Identification_Area>
      <Context_Area>
        <Investigation_Area>
          <name>MESSENGER</name>
          <type>Mission</type>
          <Internal_Reference>
            <lid_reference>urn:nasa:pds:context:investigation:mission.messenger</lid_reference>
            <reference_type>document_to_investigation</reference_type>
          </Internal_Reference>
        </Investigation_Area>
        <Observing_System>
          <name>MESSENGER</name>
          <Observing_System_Component>
            <name>MESSENGER</name>
            <type>Host</type>
            <Internal_Reference>
              <lid_reference>urn:nasa:pds:context:instrument_host:spacecraft.mess</lid_reference>
              <reference_type>is_instrument_host</reference_type>
            </Internal_Reference>
          </Observing_System_Component>
          <Observing_System_Component>
            <name>NS</name>
            <type>Instrument</type>
            <Internal_Reference>
              <lid_reference>urn:nasa:pds:context:instrument:ns.mess</lid_reference>
              <reference_type>is_instrument</reference_type>
            </Internal_Reference>
          </Observing_System_Component>
        </Observing_System>
        <Target_Identification>
          <name>Mercury</name>
          <type>Planet</type>
          <Internal_Reference>
            <lid_reference>urn:nasa:pds:context:target:planet.mercury</lid_reference>
            <reference_type>document_to_target</reference_type>
          </Internal_Reference>
        </Target_Identification>
      </Context_Area>
      <Document>
        <publication_date>2016-03-16</publication_date>
        <description>Description of the MESSENGER Neutron Spectrometer (NS).</description>
        <Document_Edition>
          <edition_name>MESSENGER NS Instrument Description</edition_name>
          <language>English</language>
          <files>1</files>
          <Document_File>
            <file_name>ns_inst.pdf</file_name>
            <document_standard_id>PDF/A</document_standard_id>
          </Document_File>
        </Document_Edition>
      </Document>
    </Product_Document>
  </pds_api:pds4>
</pds_api:product>
```

**Comment by tloubrieu-jpl** [comment]
Thanks @al-niessner ,

I am seeing that the meta content is still without the pds_api prefix. Is it complicated to update ?
Anyway I will merge it as-is and add a note in @jordanpadams ticket for this section (https://github.com/NASA-PDS/pds-api/issues/154)

**Comment by al-niessner** [comment]
> Thanks @al-niessner ,
> 
> I am seeing that the meta content is still without the pds_api prefix. Is it complicated to update ? Anyway I will merge it as-is and add a note in @jordanpadams ticket for this section ([NASA-PDS/pds-api#154](https://github.com/NASA-PDS/pds-api/issues/154))

@tloubrieu-jpl 

I thought the answer would be yes but it is no. I have nothing else going today so I will just keep working at it but it will not be done before lunch.

The jackson XmlMapper is being used to convert that part and it does not obviously accept a namespace qualifier. It must allow it somewhere so I will keep looking. 

--- more

Just googled about and jackson wants '@' attributes to add the namespace. However the bean being translated is part of the code generated by swagger. Not sure I can add them via swagger so looking for more direct method.

**Pull request merged**